presto-main/src/main/java/com/facebook/presto/sql/gen/JoinFilterFunctionCompiler.java

```diff
@@ -0,0 +1,190 @@
+import static com.facebook.presto.bytecode.expression.BytecodeExpressions.constantFalse;
+import static com.facebook.presto.sql.gen.BytecodeUtils.loadConstant;
+import static java.lang.String.format;
+
+public class JoinFilterFunctionCompiler
+{
+    private final Metadata metadata;
+
+    public JoinFilterFunctionCompiler(Metadata metadata)
+    {
+        this.metadata = metadata;
+    }
+
+    public void generateMethods(ClassDefinition classDefinition, CallSiteBinder callSiteBinder, RowExpression filter, int leftBlocksSize)
+    {
+        CachedInstanceBinder cachedInstanceBinder = new CachedInstanceBinder(classDefinition, callSiteBinder);
+        generateFilterMethod(classDefinition, callSiteBinder, cachedInstanceBinder, filter, leftBlocksSize);
+        generateConstructor(classDefinition, cachedInstanceBinder);
+    }
+
+    private void generateFilterMethod(ClassDefinition classDefinition, CallSiteBinder callSiteBinder, CachedInstanceBinder cachedInstanceBinder, RowExpression filter, int leftBlocksSize)
+    {
+        // todo handle TRY expression
+        Map<CallExpression, MethodDefinition> tryMethodMap = ImmutableMap.of();
+
+        //int leftPosition, Block[] leftBlocks, int rightPosition, Block[] rightBlocks
+        Parameter leftPosition = arg("leftPosition", int.class);
+        Parameter leftBlocks = arg("leftBlocks", Block[].class);
+        Parameter rightPosition = arg("rightPosition", int.class);
+        Parameter rightBlocks = arg("rightBlocks", Block[].class);
+
+        MethodDefinition method = classDefinition.declareMethod(
+                a(PUBLIC),
+                "filter",
+                type(boolean.class),
+                ImmutableList.<Parameter>builder()
+                        .add(leftPosition)
+                        .add(leftBlocks)
+                        .add(rightPosition)
+                        .add(rightBlocks)
+                        .build());
+
+        method.comment("filter: %s", filter.toString());
+        BytecodeBlock body = method.getBody();
+
+        Scope scope = method.getScope();
+        Variable wasNullVariable = scope.declareVariable("wasNull", body, constantFalse());
+
+        BytecodeExpressionVisitor visitor = new BytecodeExpressionVisitor(
+                callSiteBinder,
+                cachedInstanceBinder,
+                fieldReferenceCompiler(callSiteBinder, leftPosition, leftBlocks, rightPosition, rightBlocks, leftBlocksSize, wasNullVariable),
+                metadata.getFunctionRegistry(),
+                tryMethodMap);
+
+        BytecodeNode visitorBody = filter.accept(visitor, scope);
+
+        Variable result = scope.declareVariable(boolean.class, "result");
+        body.append(visitorBody)
+                .putVariable(result)
+                .append(new IfStatement()
+                        .condition(wasNullVariable)
+                        .ifTrue(constantFalse().ret())
+                        .ifFalse(result.ret()));
```

Comment:
It is initialized.
`visitorBody` leaves computation result on stack and then it is put into `result` variable via `.putVariable(result)`.